### ISSUE TITLE
Mining to NiceHash: Job not found, Invalid nonce

### ISSUE THREAD

**Issue by borzaka** [fireice-uk/xmr-stak#835]
After mining ~16 hours, this is my result:
```
RESULT REPORT
Difficulty       : 400015
Good results     : 1142 / 1178 (96.9 %)
Avg result time  : 48.5 sec
Pool-side hashes : 366263402

Top 10 best results found:
|  0 |       2804897726 |  1 |        241975830 |
|  2 |        230217641 |  3 |        210438747 |
|  4 |        186649657 |  5 |        167523502 |
|  6 |        102330571 |  7 |         81265676 |
|  8 |         46654453 |  9 |         42651696 |

Error details:
| Count | Error text                       | Last seen           |
|    27 | Job not found.                   | 2018-01-08 13:20:05 |
|     9 | Invalid nonce; is miner not comp | 2018-01-08 12:57:09 |
```

# My config

config.txt:
```
"pool_list" :
[
	{"pool_address" : "stratum+tcp://cryptonight.eu.nicehash.com:3355", "wallet_address" : "3B3r1yBE8ub2kaewHAieLbfL1hfv6bNGj1.vegaworker", "pool_password" : "x", "use_nicehash" : true, "use_tls" : false, "tls_fingerprint" : "", "pool_weight" : 3 },
	{"pool_address" : "stratum+tcp://cryptonight.usa.nicehash.com:3355", "wallet_address" : "3B3r1yBE8ub2kaewHAieLbfL1hfv6bNGj1.vegaworker", "pool_password" : "x", "use_nicehash" : true, "use_tls" : false, "tls_fingerprint" : "", "pool_weight" : 2 },
	{"pool_address" : "stratum+tcp://cryptonight.br.nicehash.com:3355", "wallet_address" : "3B3r1yBE8ub2kaewHAieLbfL1hfv6bNGj1.vegaworker", "pool_password" : "x", "use_nicehash" : true, "use_tls" : false, "tls_fingerprint" : "", "pool_weight" : 1 },
],
```
amd.txt
```
"gpu_threads_conf" : [ 
    { "index" : 0, "intensity" : 1932, "worksize" : 8, "affine_to_cpu" : 1, "strided_index" : true },
    { "index" : 0, "intensity" : 1932, "worksize" : 8, "affine_to_cpu" : 1, "strided_index" : true },
    { "index" : 1, "intensity" : 1932, "worksize" : 8, "affine_to_cpu" : 1, "strided_index" : true },
    { "index" : 1, "intensity" : 1932, "worksize" : 8, "affine_to_cpu" : 1, "strided_index" : true },
    { "index" : 2, "intensity" : 1800, "worksize" : 8, "affine_to_cpu" : 1, "strided_index" : true },
    { "index" : 2, "intensity" : 1800, "worksize" : 8, "affine_to_cpu" : 1, "strided_index" : true },
],
```
cpu.txt
```
"cpu_threads_conf" :
[
    { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 0 },
    { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 2 },
    { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 4 },
    { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 6 },
    { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 8 },
    { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 10 },
    { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 12 },
    { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 14 },
],
```

# Basic information
  - 3x Vega 64
  - Ryzen 7 1700X
  - ~6500 H/s
  - xmr-stak version: `xmr-stak/2.2.0/c4400d19/master/win/nvidia-amd-cpu/aeon-monero/20`
  - Blockchain driver
  - Windows 10 Version 1709
  - CPU is overclocked, GPUs are overclocked and undervolted, but no AMD Invalid Results

**Comment by JerichoJones** [comment]
This is related to poor performance of Nicehash proxies. Contact them and let them know the issue that you are having with their pool.

**Comment by 2001blader** [comment]
Try XMR-Cast, someone else reported that as a fix earlier. If that works fine, then that's proof that it is a stak problem, and not a nicehash one.

**Comment by toekyoe** [comment]
I have the same issue.  I went to https://www.nicehash.com/support and submitted a ticket.  If someone tries XMR-cast and does not have an issue, please let us know.

**Comment by Zazax21** [comment]
Guys, Hi, i have the same problem as mentioned above, i'm mining with with 5 x vega 64 and sometimes I have this problem
Hash rate dropped from 10249.0 H/s to  H/s

**Comment by JerichoJones** [comment]
>Hash rate dropped from 10249.0 H/s to H/s
Not a STAK error. JJs hash monitor is not supported.

**Comment by Zazax21** [comment]
What do you mean mate ?

**Comment by borzaka** [comment]
Your problem is probably caused by beta blockchain driver and/or too much overclock. The blockchain driver is very sensitive, especially with Vega. Common issue, I have it too, but not XMR-Stak related.

**Comment by prettycode** [comment]
Have you found a resolution, @borzaka? Have same issue of many, many "Job not found" on my five-Vega rig. Miner reporting 9300 H/s, but after all these "Job not found" errors, calculated hashrate is in the 7,000s. Yikes.

**Comment by borzaka** [comment]
I have also submitted a ticket to NiceHash.
This is 100% NiceHash related. There is no problem mining to another CryptoNight pool with the same settings.

**Comment by kyleboddy** [comment]
While this is Nicehash related, it is also xmr-stak related.

![image](https://user-images.githubusercontent.com/746351/35146779-e37e5934-fcc0-11e7-945a-796482612689.png)

These are my 6xVega56 statistics and tomorrow I'll upload an evening of castXMR's data, but I guarantee it will be less than 4% in lost jobs AND devfee combined. castXMR patched and made available the `--fastjobswitch` option which solved the vast majority of these issues (some small number of "job not found" errors will occur on any mining software due to bad luck and high difficulty of Nicehash).

**Comment by psychocrypt** [comment]
I will add fastswitch soon. But the erros of the nicehash pool can not
fixed within the miner.

**Comment by ChromeHearts** [comment]
I have ran both XMR-STAK & CAST XMR for over 24 hours.  CAST has an option --fastjobswitch mode that fixes the stale job errors.  Error rate dropped ~1%.  Without the option, CAST & STAK have similar rate.
XMR-STAK had 4~6% error rate, similar to other users here.

@psychocrypt Looking forward to the fast switch mode!  Thank you in advance!

**Comment by kyleboddy** [comment]
Here's the results of running castXMR overnight, as promised:

![image](https://user-images.githubusercontent.com/746351/35157528-1a66122a-fce9-11e7-9caa-db95c7ab4726.png)

`Fastjobswitch` would be a huge addition. Thank you!

**Comment by kyleboddy** [comment]
![image](https://user-images.githubusercontent.com/746351/35178054-0437993c-fd3a-11e7-9f07-a31ca4c43be4.png)

Just contributing more data - longer runtime, same machine, castXMR output.

**Comment by JerichoJones** [comment]
Now compare the data against what the pool reports.

**Comment by borzaka** [comment]
Reply from NiceHash support:
> Hello!
> 
> We suggest you to use our official NiceHash Miner, which you can download from here: https://miner.nicehash.com
> 
> All other variants of mining software may not be compatible with NiceHash due to some protocol specifics.
> 
> Kind regards,
> NiceHash Team

**Comment by prettycode** [comment]
Switched to Cast XMR with --fastjobswitch and getting much, much fewer invalid results. Pool-side/calculated hashrate is about 4 - 6% higher now vs. with xmr-stak and all the "Job not found." No choice but to switch to Cast with my 6-Vega rig. :(

**Comment by xkdz** [comment]
same here.. went from ~20-25% failure rate that I was getting with XMR-Stak down to 99%+ with Cast... something is up, it only started recently too..  Problems in 2 vega rigs with stak :(

**Comment by JoKeRz42o** [comment]
@borzaka I've been using the official NiceHash Miner (NHML 1.8.2.0) and honestly is as bad if not worse than the latest version of most of the miners included with your software. 

On average I'm showing 4-5% errors on the miner side...

```
Difficulty       : 400015
Good results     : 2313 / 2405 (96.2 %)
Avg result time  : 22.3 sec
Pool-side hashes : 5400191

Top 10 best results found:
|  0 |       1821927604 |  1 |        358736341 |
|  2 |        177323025 |  3 |        174301948 |
|  4 |        165253070 |  5 |        140589606 |
|  6 |        137797074 |  7 |         88931207 |
|  8 |         82396103 |  9 |         81954457 |

Error details:
| Count | Error text                       | Last seen           |
|    85 | Job not found.                   | 2018-01-24 23:39:15 |
|     6 | Invalid nonce; is miner not comp | 2018-01-24 10:00:46 |
|     1 | AMD Invalid Result               | 2018-01-24 16:34:25 |
```

##**...but what's even worse is that on the pool side the #'s are even more dreadful...**

![image](https://user-images.githubusercontent.com/33408283/35377338-94f3f43e-0163-11e8-9da1-107a41c36bee.png)

As you can see, multiple times throughout the day my profitability is down to ZERO (0.0000) BTC. Mind you, usually i get 5+ of those 0 points... 2-3 the last 24hours is a bit of a blessing. 

I've contacted customer support three (3) times over the course of 2 weeks with one reply back saying they would look into it and get back to me.

**Comment by DrStein99** [comment]
Hi.  I have been using the software with nicehash too.  I was also getting those dips.  I would get periods of time where transfers would drop low, and I check my log with series of disconnects over the course of 45 mintues some times.

I tried changing my software so many times in 2 weeks, I can not count.  I got frustrated so I explored  to use xmr-stak independantly without the nicehash front end.

I ran tests on my smaller machines connecting to other various xmr monero pools, to compare.  I wanted to see how their system statistics -vs- my internal system.  When xmr-stak is run on servers that are NOT nicehash; I do NOT get the drop-valleys, reports of socket disconnects, and/or periods long unusual latent pool has accepts.

The "average result time" on every other xmr pool I have tested so far was below 30 seconds.  "average result time" for all my higher end machines is well over 130 seconds, and sometimes 200, 300, seconds.  When I find 500+ seconds I reset xmr-stak, and try and find the problem.  I try to analyze but have nothing conclusive - besides a theory for disruption in service from my site to the nicehash server pool.

**Comment by kyleboddy** [comment]
"Average result time" has everything to do with the difficulty. Nicehash's static difficulty on CryptoNight starts at 200,000 which is well, WELL above vardiff for other pools. This is why it takes very long to get a result, and why sometimes when you do, you submit a bad share. This is very likely related to fast job switching, not Nicehash directly.

I have validated that castXMR poolside submits all clean shares with no issues when `--fastjobswitch` is enabled, but xmr-stak has the same problem castXMR used to have before that option existed (3-5% stale shares).

**Comment by kyleboddy** [comment]
![image](https://user-images.githubusercontent.com/746351/35384567-81431a16-017a-11e8-824b-acad43909ef1.png)

Recent mining details using castXMR on a long run.

**Comment by psychocrypt** [comment]
#977 should solve the outdated block issue. I currently running my miner against the nicehash pool to test if I can see invalid shares.

**Comment by psychocrypt** [comment]
for all nice hash miner #978 is also interesting. We had a very small nonce overlap. This results in a very very small possibility that duplicated shares can be created.

**Comment by kyleboddy** [comment]
Thanks, will test on the next commit.

**Comment by minerbird** [comment]
I have the same issue (`Job not found` errors around 5%). I've been mining on NiceHash's cryptonight EU pool but tested other NiceHash pools with the same or worse results. I do believe this is partly an issue on NiceHash's end because I am experiencing a lot of disconnects (socket errors) while using a NiceHash pool. When mining ETN or monero on a nanopool server I don't get stale jobs or experience any disconnects.

As @kyleboddy already pointed out:
> I have validated that castXMR poolside submits all clean shares with no issues when --fastjobswitch is enabled, but xmr-stak has the same problem castXMR used to have before that option existed (3-5% stale shares).

Is there a way to implement this setting (from castXMR) into xmr-stak? I had hoped that #977 or #978 would help solve this issue but #977  is closed and it looks like #978 will be closed soon.

**Comment by toekyoe** [comment]
Has anyone had success with the code changes in #977 or #978?

**Comment by borzaka** [comment]
I'am currently using the latest dev branch, compiled on yesterday (01.26.).

After ~8 hours of mining to NiceHash:
```
RESULT REPORT
Difficulty       : 200007
Good results     : 612 / 635 (96.4 %)
Avg result time  : 48.2 sec
Pool-side hashes : 198207386

Top 10 best results found:
|  0 |        219852473 |  1 |        192800281 |
|  2 |        124500090 |  3 |         90206818 |
|  4 |         56914121 |  5 |         49705601 |
|  6 |         34597221 |  7 |         32560097 |
|  8 |         29064942 |  9 |         28624119 |

Error details:
| Count | Error text                       | Last seen           |
|    21 | Job not found.                   | 2018-01-27 12:30:51 |
|     2 | AMD Invalid Result GPU ID 1      | 2018-01-27 06:51:59 |
```

No more Invalid nonce, only Job not founds, but it's way more than with the latest release version (~16 hours, 27 Job not found).
`AMD Invalid Result GPU ID 1` is because of too much overclock, I know.

**Comment by minerbird** [comment]
Compiled the latest version from dev branch which includes #996 today. xmr-stak running for about 9 hours now without "invalid nonce" errors. Thank you for your hard work @psychocrypt & @fireice-uk.
```
RESULT REPORT
Difficulty       : 200007
Good results     : 406 / 417 (97.4 %)
Avg result time  : 98.1 sec
Pool-side hashes : 65602296

[...]

Error details:
| Count | Error text                       | Last seen           |
|    9  | Job not found.                   | 2018-01-30 19:54:34 |
|    2  | AMD Invalid Result GPU ID 0      | 2018-01-30 19:36:25 |
```

**Comment by borzaka** [comment]
I'am still having Invalid nonce error with the latest dev branch compiled on yesterday (01.30.):

After ~7,5 hours mining to NiceHash:
```
HASHRATE REPORT - AMD
| ID |    10s |    60s |    15m | ID |    10s |    60s |    15m |
|  0 | 1023.3 | 1022.2 | 1021.3 |  1 | 1021.0 | 1022.0 | 1021.3 |
|  2 | 1016.8 | 1021.1 | 1022.8 |  3 | 1026.7 | 1022.6 | 1022.9 |
|  4 |  939.5 |  939.6 |  939.5 |  5 |  936.4 |  939.2 |  939.5 |
Totals (AMD):  5963.7 5966.7 5967.3 H/s
-----------------------------------------------------------------
Totals (ALL):   6545.4 6590.5 6598.4 H/s
Highest:  6625.0 H/s
-----------------------------------------------------------------
RESULT REPORT
Difficulty       : 400015
Good results     : 545 / 559 (97.5 %)
Avg result time  : 48.4 sec
Pool-side hashes : 159806669

Top 10 best results found:
|  0 |        946095324 |  1 |        451201395 |
|  2 |         85282705 |  3 |         39346610 |
|  4 |         33953383 |  5 |         29997236 |
|  6 |         26507963 |  7 |         22740497 |
|  8 |         22330262 |  9 |         19314346 |

Error details:
| Count | Error text                       | Last seen           |
|     1 | Invalid nonce; is miner not comp | 2018-01-31 09:32:54 |
|    13 | Job not found.                   | 2018-01-31 16:26:50 |
```

**Comment by minerbird** [comment]
Me too but I get way fewer errors than before.

**Comment by mardicas** [comment]
I am getting also fewer errors from the dev version than the last release. Running for ~12 hours now. None of them are Invalid nonce; Only "RECEIVE error: socket closed". Also from nicehash UI seems the rate has imrpoved and does not drop as much.

**Comment by hchan123** [comment]
I'm running the latest dev.  Here's my results from a 12 hour run.  Getting a ton of job not founds and 1 nonce error.  Not sure if job not founds are poolside related.

RESULT REPORT
Difficulty       : 200007
Good results     : 546 / 574 (95.1 %)
Avg result time  : 72.3 sec
Pool-side hashes : 84002947

Top 10 best results found:
|  0 |         39347943 |  1 |         26166618 |
|  2 |         22577700 |  3 |         19944564 |
|  4 |         19102979 |  5 |         18865856 |
|  6 |         17145834 |  7 |         13023604 |
|  8 |         11178336 |  9 |         10409639 |

Error details:
| Count | Error text                       | Last seen           |
|     7 | [NETWORK ERROR]                  | 2018-02-01 11:06:28 |
|    20 | Job not found.                   | 2018-02-01 19:29:26 |
|     1 | Invalid nonce; is miner not comp | 2018-02-01 10:55:24 |

**Comment by minerbird** [comment]
@hchan123 What pool are you connected to? Getting a lot of "Job not found" and connection errors as well. I am connected to `cryptonight.eu.nicehash.com:33355`.

**Comment by borzaka** [comment]
@minerbird Check the issue Title. We are mining to NiceHash with this XMR-Stak CryptoNight miner.

After ~13 hours:
```
Error details:
| Count | Error text                       | Last seen           |
|    24 | Job not found.                   | 2018-02-03 21:32:55 |
|     2 | Invalid nonce; is miner not comp | 2018-02-03 12:20:33 |
```

**Comment by Sandown** [comment]
I also have this issue, although I'm running v2.0.0, it doesn't look like its been resolved in the latest dev versions. I'm wondering if there's anything else we can do to help the Devs resolve this?

**Comment by psychocrypt** [comment]
I tested to mine against the eu proxy of nicehash with one rx570 and get no
error job not found.
Currently it is not possible to reproduce the issue.

**Comment by borzaka** [comment]
@psychocrypt For how many hours have you tested? Because NiceHash's CrytoNight difficulty is very high, and therefore shares are submitted very rarely; the problem will come up later on slow machines.
I have 6500 H/s, your RX 570 has around 700 H/s. Based on my statistics, you should get your first job not found after ~5 hours.

**Comment by Sandown** [comment]
Just to add, my rigs run at 16 KH/s each. Logs from one of the rigs:


Error text
--

Job not found.
150 | 2018-02-05 09:24:16
Invalid nonce; is miner not compatible with NiceHash?
6 | 2018-02-04 09:29:34


Pool address | stratum+tcp://cryptonight.eu.nicehash.com:3355
Connected since: 2018-02-04 11:00:35
ping time : 81 ms

**Comment by toekyoe** [comment]
Come on @psychocrypt.  There are a bunch of people telling you that they’re getting about 5% invalid results, myself included.

I see them in real-time in the error log.  It always happens after a new job is detected.  And the first result submitted after that sometimes is an invalid result.  To my novice self, it seems like the mining software isn’t switching the new job fast enough, as it’s still trying to submit shares on the old job.  Thoughts?

**Comment by hchan123** [comment]
@minerbird  I usually connect to cryptonight.usa..nicehash.com:33355 but also have cryptonight.eu.nicehash.com:33355 and cryptonight.uk.nicehash.com:33355 as failover.

**Comment by psychocrypt** [comment]
After several hours I can now reproduce the issue. As some of you wrote the reason is that a share is send to the pool during or short after a new job is arriving. 
Normally this is shown as expired block but nicehash is reporting it under an not common error.
The point is that there is no reason why the share of the previous job should not be valid and payed. This is only the case for pools with bad software. 

@toekyoe  The miner is not switching the job to slow. The reason for this "error" is that the last bunch of hashes is calculated on the gpu. There are possible solutions which throw away good hashes but will reduce this "error"

- 1. cancel the current calculated bunch of hashes if a new job arrived
   - The GPU is running 90% of the time (~2sec) one kernel, if we cancel the current job after the kernel and start direct with the new job we will only emulate good hashes without any benefit in time. This is what "fastswitch" in `castxmr` is doing.
- 2. do not submit late valid shares
  - This means that on well programmed pools where also the previous job hashes are paid will get less hashes and therefore the miner lose money.

What I will say, we can avoid this "error" in xmr-stak but this will be only a visual effect and will be equal with cheating ( it only beautify the statistics). 

The argument that `castxmr` is not having this issue is not correct: Nobody knows how the error counting is going on in `castxmr` and the `fastswitch` option is only an option a throw away good hashes.

**If someone will open an POLL where the community can vote if the fast switch option should be implemented in `xmr-stak` or not I will discuss it with @fireice-uk again.**

**Comment by jrodrigog** [comment]
It seems to me that _nicehash_ have changed something around 20180210 18:00:00 CET.

Before I had around 2% "Job not found" **plus** some (a few) "Invalid nonce".
Now I have **only** "Job not found" errors but it is a solid 4%. I'm doing 24 hours runs.

To me this change is clear on the _nicehash_ plots; maybe it is something local to my setup maybe not.

**Comment by vans163** [comment]
"cancel the current calculated bunch of hashes if a new job arrived
The GPU is running 90% of the time (~2sec) one kernel, if we cancel the current job after the kernel and start direct with the new job we will only emulate good hashes without any benefit in time. This is what "fastswitch" in castxmr is doing."

@psychocrypt This part does not make sense to me "only emulate good hashes without any benefit in time", how can you emulate a good share? Either the share is good, or its bad. Emulating a bad share seems more plausible.

What I do not understand is, canceling the current job would still result in a ~2 sec run but no submitted stale share?  Or cancelling the current job would drop the current work and immediately switch (at the cost of losing at most ~2 seconds on a stale share, average 1 second). But the logic is that stale share will be paid for by the pool.

So the use cases, correct me if I am wrong.
#1 pool pays for stale shares, job switch makes us spend between 0-2 seconds running to completion, average 1 second gained mining per job. This is 3.33% gained hashrate on a 30s coin.

#2 pool does not pay stale shares, job switch makes us spend between 0-2 seconds running to completion, average 1 second lost per job. This is 3.33% lost hashrate on a 30s coin.

#3 pool does not pay stale shares, immediate job switch would make us drop current work, losing 0-2 seconds, average 1 second, we dont get paid for stale share anyways. This is a 3.33% gain in hashrate compared to #2 (net 0.00 difference), as we get our real hashrate as miner reports at pool side

**Comment by borzaka** [comment]
To switch to new job with Cast XMR is about ~20-40 ms in my case:
```
[18:04:44] New job received. Avg Job Time: 49.0 sec
[18:04:44] GPU0: 26 ms needed to switch to new job
[18:04:44] GPU2: 34 ms needed to switch to new job
[18:04:44] GPU1: 37 ms needed to switch to new job
```

Not seconds!

> - --fastjobswitch option (experimental) to force fast switching to new jobs. When enabled the GPUs will switch to the new job within less then 100ms. This improves the effective hash rate by 1 to 5% as nearly no GPU performance will be wasted on calculating outdated shares anymore. The drawback is a slightly lower displayed hashrate as while switching no hashes are calculated.

**Comment by fireice-uk** [comment]
One of those days I swear I will put an option that simply does hashrate = hashrate * 1.10 like castxmr

Until then, if you don't know why pool **needs** stale shares, please learn what an orphaned block is and how mining works.

**Comment by vans163** [comment]
@fireice-uk orphaned block is most of the time the result of a valid solved block that did not propagate properly.  It could also be the result of a 51% attack where total nodes in the network are rejecting your valid blocks from the main chain and are solving faster their own blocks due to hash power which are getting accepted by >51% of the network.

A stale share could lead to solving two blocks at the same height, then the highest difficulty share would be taken as valid in most cases (depends on blockchain).

Could you enlighten us how mining works then? Because I think some of us are not seeing the connection of how stale shares are useful to the underlying block chain.

**Comment by fireice-uk** [comment]
@vans163 

1. Miner GPU starts a batch on block 10
2. A new job arrives from a pool for block 11
3. Miner GPU finds a solution to block 10, submits as a stale share
4. Since the solution's diff > block 11 diff, it becomes the new block 11, orphaning the old one

**Comment by borzaka** [comment]
Another report after ~25 hours.

```
RESULT REPORT
Difficulty       : 200007
Good results     : 1812 / 1859 (97.5 %)
Avg result time  : 47.6 sec
Pool-side hashes : 93503775

Top 10 best results found:
|  0 |        414940385 |  1 |        259847514 |
|  2 |        246063839 |  3 |        239841035 |
|  4 |        118868302 |  5 |         91313547 |
|  6 |         81895170 |  7 |         53190579 |
|  8 |         39688135 |  9 |         38225717 |

Error details:
| Count | Error text                       | Last seen           |
|    37 | Job not found.                   | 2018-03-06 20:44:21 |
|    10 | Invalid nonce; is miner not comp | 2018-03-06 19:27:38 |
```
(Dev branch compiled on 01.30.)

With ~6600 H/s

**Comment by vans163** [comment]
@CircusDad It occurs everytime a new block arrives, what your seeing as hard evidence from comment #835 is when an actual share was found, AND got rejected as stale. Don't forget you still lose all that hashing time mining stale block, its just if you did not find a share, it is never reported on 
the console printout. So you don't have hard evidence from the miner logs. Perhaps don't rely on logs and try using dtrace for your evidence (assuming your on linux).

Tl;Dr; The lost hashing time on a stale block is ONLY logged if an actual stale share is found (which depending on difficulty and hashrate could be every 100-100000 times). So multiply what your seeing by 100-1000000 times.

```
Since (block 11' diff + total_chain_diff) > (block 11 diff + total_chain_diff) chain with block 11' is longer and the rest of the network reorgs accordingly.
```

Yea.. your stale share just caused an orphan block for some pool somewhere, now its just an asshole move to mine stale shares.  Your causing orphan blocks for legit pools that are small and community driven.

**Comment by JerichoJones** [comment]
I need more popcorn.

**Comment by fireice-uk** [comment]
> Your causing orphan blocks for legit pools that are small and community driven.

:'( I will let you donate your hashrate to a mom&pop pool like nicehash then. 

On a more serious note, do you know a saying "Better to remain silent and be thought a fool than speak and remove all doubt."  

Sorry, I just can't say it with a straight face, "mom&pop community owned pool". LOL.  Thanks for brightening up my evening m8.

**Comment by CircusDad** [comment]
I stand corrected.  Being reported as a "Job not found" means that the miner found the hash of proper difficulty (share) during that one second and thus had the whole share rejected.  Per @borzaka's [stats](https://github.com/fireice-uk/xmr-stak/issues/835#issuecomment-370905421), it _probably_  took him about 46 seconds on average to find that hash and it will _probably_ be another 46 seconds before he finds another.  

He only spent 1 second finishing his last assignment but, depending how you do the 'sunk cost' probability math, he lost not 1 second of mining credit but rather ~46 seconds of mining credit (his average share time)

To use @borzaka [latest results](https://github.com/fireice-uk/xmr-stak/issues/835#issuecomment-370905421), he had 53 "job not found" in 2068 shares at an average share time of 46 seconds.

(53 * 46 seconds lost credit) / ( 2068 * 46 mining seconds ) = 2.6% lost mining credit... 

That is a bit more significant then I stated [before ](https://github.com/fireice-uk/xmr-stak/issues/835#issuecomment-370657676).  

I happen to agree with @fireice-uk that a pool should credit for all assigned work that is performed.  But I do think that since we know that nicehash doesn't, perhaps it is worth a "FastJobSwitch" feature that automatically becomes active as part of what it means when "nicehash" is set to true.

**Comment by JerichoJones** [comment]
IOW, Nicehash blows.

**Comment by fireice-uk** [comment]
```But I do think that since we know that nicehash doesn't, perhaps it is worth a "FastJobSwitch" feature that automatically becomes active as part of what it means when "nicehash" is set to true.```

I would have two objections to that:

- It gimps the performance for users that use this mode and don't use nicehash (proxies)
- It sets a precedent that we will try to work around poorly coded pools. 

My prediction is that nicehash will drop support for Monero anyway, so this whole debate may be moot.
